docs/src/main/asciidoc/building-my-first-extension-guide.adoc

```diff
@@ -0,0 +1,684 @@
+-------------
+
+Looks good!! Congratulations you just finished your first extension.
+
+=== Debugging your extension
+
+_If debugging is the process of removing bugs, then programming must be the process of putting them in._
+Edsger W. Dijkstra
+
+
+*Debugging your application build*
+
+Since your extension deployment is made during the application build, this process is triggered by your build tool.
+That means if your want to debug this phase you need to launch your build tool with the remote debug mode switched one.
+
+_Maven_
+
+You can activate Maven remote debugging by using `mvnDebug`. You can launch your application with the following command line:
+
+`mvnDebug clean compile quarkus:dev`
+
+By default, Maven will wait for a connection on `localhost:8000`. Now, you can run your IDE `Remote` configuration to attach it to `localhost:5000`.
+
+_Gradle_
+
+TODO
+
+*Debugging your test extension*
+
+We have seen together how to test your extension and sometimes things don't go so well and you want to debug your tests.
+Same principle here, the trick is to enable the Maven Surfire remote debugging in order to attach an IDE `Remote` configuration.
+
+[source, shell]
+-------------
+$ cd `./greeting`
+$ mvn clean test -Dmaven.surefire.debug
+-------------
+
+By default, Maven will wait for a connection on `localhost:5005`.
+
+=== Extension publication
+
+Now that you just finish to build your first extension you should be eager to use it in a Quarkus application! 
+
+*Classic Maven publication*
+
+Because your extension produces a classic JAR, the easiest way to share your extension is to publish it to a Maven repository.
+Once published you can simply declare it with your project dependencies
+```
+...
+    <dependencyManagement>
+        <dependencies>
+            <dependency>
+                <groupId>${quarkus.platform.group-id}</groupId>
+                <artifactId>${quarkus.platform.artifact-id}</artifactId>
+                <version>${quarkus.platform.version}</version>
+                <type>pom</type>
+                <scope>import</scope>
+            </dependency>
+        </dependencies>
+    </dependencyManagement>
+    <dependencies>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-resteasy</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.acme</groupId>
+            <artifactId>quarkus-greeting</artifactId>
+        </dependency>
+    </dependencies>
+...
+```
+
+
+From a extension developper standpoint this publication strategy is very handy and fast but Quarkus wants to go one step further by also ensure a reliability of the ecosystem for the people who will use the extensions.
+Think about it, we all had a poor Developper eXperience with an unmaintained library, a incompatibility between dependencies (and we don't even talk about legal issues).
+That's why there is the Quarkus Platform.
+
+*Quarkus Platform*
```

Comment:
The first entry point into the extension ecosystem will be Quarkus Universe, which aggregates extensions from the supported platform and the rest of the ecosystem. It's basically a catalog of all the known extensions. Our project manipulating tooling (create project, list extensions, add extensions) are using the universe as the source of extensions.

The platform is a subset of the universe that we selected and guarantee a certain level of quality for (besides the technological criteria).

Reply:
Not sure if you proposes a replacement for the sentence or if you want me to add additionnal comments?

Reply:
Sorry for the confusion. We actually have not released `quarkus-platform-bom` yet. We currently have `quarkus-universe-bom` which will be replaced by `quarkus-platform-bom` at some point.